### PR TITLE
Feature/merge GitHub resolve func issue

### PR DESCRIPTION
**Pull request by gabrielrutsch**
This PR includes a tighter integration of the UART control IFC to the anasymod control API.

Also xsct support was integrated tighter to the anasymod configuration layer.

Regression tests are now compatible to IFX regression infrastructure.

### CODE REVIEW

**Comment by sgherbst** [comment]
@gabrielrutsch thanks for the PR!  I'm not sure if you have access to the BuildKite messages, but it appears that the test ``unittests/basic_sim/test_basic_sim.py::TestBasicSIM::test_firmware`` is failing with the following error:

```text
>       self.vid_list = self.pcfg.board.uart_zynq_vid
E       AttributeError: 'ZC702' object has no attribute 'uart_zynq_vid'
anasymod/sim_ctrl/uart_zynq_ctrlapi.py:53: AttributeError
```

The BuildKite regression server uses ZC702, and I generally use ZC706 or ZCU106 locally -- would you mind checking that these boards had this new property updated?  Thanks.

File: anasymod/fpga_boards/boards.py [hunk 1]
```diff
@@ -18,9 +18,11 @@ class PYNQ_Z1(FPGA_Board):
     board_part = 'www.digilentinc.com:pynq-z1:part0:1.0'
     full_part_name = 'xc7z020clg400-1'
     short_part_name = 'xc7z020'
+    bram = 4.9e6
+    dbg_hub_clk_freq = 100e6
     fpga_sim_ctrl = [FPGASimCtrl.UART_ZYNQ, FPGASimCtrl.VIVADO_VIO]
-    uart_vid = 1027
-    uart_pid = 24592
+    uart_zynq_vid = [1027]
```
Comment (sgherbst) [inline anasymod/fpga_boards/boards.py:24]: I think this name change likely has to be updated for the other boards as well; would you mind doing that?  Thanks.

File: unittests/basic_sim/test_basic_sim.py [hunk 2]
```diff
@@ -55,21 +55,21 @@ def __init__(self):
...
-        except:
-            pass
-
-    list_files_delete = []
-    for file in list_files_delete:
-        try:
-            shutil.rmtree(os.path.join(os.path.dirname(__file__), file))
-        except:
-            pass
+    pass
```
Comment (sgherbst) [inline unittests/basic_sim/test_basic_sim.py:58]: is commented-out section intended?

**Comment by sgherbst** [comment]
Looks like a comprehensive PR, thanks for putting in the effort!  Two general comments:
1. Since there is now a distinction between "default firmware" and "custom firmware", I think we should exercise both paths in regression testing.  I think this could be done by running two versions of the firmware test -- one using the autogenerated firmware, and one using the custom firmware (which already exists, since that is how the test was previously implemented).  I think these tests should both be run on the FPGA board.  Note that ``regress.sh`` would need to be updated to call the new variant of ``test_firmware`` under the ``emulate_vivado`` tests.
2. Just wanted to check that it is still possible to rebuild only the firmware ELF, not the bitstream.  This is mainly relevant when developing custom firmware, since the ELF build time is much shorter than the bitstream build time.  I noticed that ``build_firmware`` and ``program_firmware`` were removed, so it was not clear to me how to do this.

**Comment by sgherbst** [comment]
@gabrielrutsch thanks for the update.  Looks like there is a remaining issue having to do with ``pid_list`` or ``vid_list``:
```text
>               if ((port.vid in self.vid_list) and (port.pid in self.pid_list)):
E               TypeError: argument of type 'int' is not iterable
```
I would guess that these are expected to be lists, but one or both are integers.

**Comment by sgherbst** [comment]
@gabrielrutsch thanks for this update.  Looks like the test progressed further, but then failed with a new error:
```text
ERROR: [HDF 64-5] File not found - /path/to/anasymod/unittests/basic_sim/build/emulate_vivado/test_firmware/fpga/prj/prj.runs/impl_1/top.hdf
ERROR: [Hsi 55-1451] Error: running open_hw_design.
ERROR: [Common 17-39] 'hsi::open_hw_design' failed due to earlier errors.
 
    while executing
"::hsi::open_hw_design $params(hw)"
    (procedure "loadhw" line 65)
    invoked from within
"loadhw -hw "/path/to/anasymod/unittests/basic_sim/build/emulate_vivado/test_firmware/fpga/prj/prj.r..."
    (file "/path/to/anasymod/unittests/basic_sim/build/emulate_vivado/test_firmware/fpga/program.tcl" line 15)
```
(``/path/to`` is the absolute location of the ``anasymod`` installation)

**Comment by gabrielrutsch** [comment]
Hi Steven,

jap, it is still failing and I think it is due to a change that I had to
implement to get it up and running on vivado 2018.2. So I will just add a
contitional statement to deal with this properly after the DVCon
presentation tomorrow xD.

Also, I added the custom_firmware test locally, which I will push tomorrow.

Cheers,
Gabriel

Steven Herbst <notifications@github.com> schrieb am Di., 27. Okt. 2020,
18:04:

**Comment by sgherbst** [comment]
@gabrielrutsch great, thanks!

**Review by sgherbst — APPROVED** [review]
LGTM, thanks!  I'll go ahead and merge this.

**Pull request merged**